test/e2e/gateway/gateway_kubernetes.go

```diff
@@ -237,6 +273,13 @@ spec:
 				client.FromKubernetesPod(ClientNamespace, "gateway-client"))
 		})
 
+		It("should proxy TCP connections", func() {
+			ProxyTcpRequest(cluster, "request", "response",
+				net.JoinHostPort(GatewayAddress("edge-gateway"), "8081"),
+				client.FromKubernetesPod(ClientNamespace, "gateway-client"),
+			)
+		})
```

Comment:
Can we maybe create a separate task for migrating gateway E2E tests to the shared environment?
Gateway E2E test are already really long-running because of `BeforeEach` everywhere.

CC: @lahabana

Reply:
https://github.com/kumahq/kuma/issues/4520